src/Engine.sol

```diff
@@ -133,20 +137,22 @@ contract Engine is
         address _sUSDProxy,
         address _pDAO,
         address _zap,
+        address payable _pay,
         address _usdc,
         address _weth
     ) {
         if (
             _perpsMarketProxy == address(0) || _spotMarketProxy == address(0)
                 || _sUSDProxy == address(0) || _zap == address(0)
-                || _usdc == address(0) || _weth == address(0)
+                || _pay == address(0) || _usdc == address(0) || _weth == address(0)
         ) revert ZeroAddress();
 
         PERPS_MARKET_PROXY = IPerpsMarketProxy(_perpsMarketProxy);
         SPOT_MARKET_PROXY = ISpotMarketProxy(_spotMarketProxy);
 
         SUSD = IERC20(_sUSDProxy);
         zap = Zap(_zap);
+        pay = Pay(_pay);
```

Comment:
mightve saved a lot of time with setup If you just deployed `Pay.sol` from the constructor btw